tests/test_django_compat.py

```diff
@@ -3,19 +3,29 @@
 
 from field_audit.models import AuditEvent
 
-from .models import SimpleModel
+from .models import ModelWithValueOnSave, SimpleModel
 
 
 class TestAuditedDbWrites(TestCase):
 
     def test_model_delete_is_audited(self):
         self.assertNoAuditEvents()
-        instance = SimpleModel.objects.create()
+        instance = SimpleModel.objects.create(id=0, value='test')
+        AuditEvent.objects.all().delete()  # delete the create audit event
+        instance.delete()
+        self.assertAuditEvent(
+            is_delete=True,
+            delta={"id": {"old": 0}, "value": {"old": 'test'}},
+        )
+
+    def test_model_delete_with_value_on_save_is_audited(self):
+        self.assertNoAuditEvents()
+        instance = ModelWithValueOnSave.objects.create(id=0)
         AuditEvent.objects.all().delete()  # delete the create audit event
         instance.delete()
         self.assertAuditEvent(
             is_delete=True,
-            delta={"id": {"old": instance.id}, "value": {"old": None}},
```

Comment:
Nice find. This test was problematic from the start (should have always used an explicit `id` and checked for the static value rather than the instance property).